### PR TITLE
[ISSUE #696] Add metrics plugin

### PR DESCRIPTION
**Pull request by ruanwenjun**
1. Add Metrics plugin module
2. Implement opentelemetry metrics module

related issue #696

### Modifications

Add metrics-api module, if user want to add other metrics exporter, they only need to add a new plugin, and implement with MetricsRegistry.java.

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@xwm1992 @Roc-00 Take a look if you are interested in.

**Review by Alonexc — DISMISSED** [review]

**Comment by qqeasonchen** [comment]
Will trace have another plugin too？

**Comment by ruanwenjun** [comment]
> Will trace have another plugin too？

No yet, the trace code is still in runtime module now.

The current trace implementation is using API to export. I think a better way is using java agent to do this work, so EventMesh don't need to export trace in code. This may need to discuss.

If we also want to provide SPI trace, we can design the trace plugin interface, but I think this can be another PR.

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/config/CommonConfiguration.java [hunk 1]
```diff
@@ -166,14 +164,22 @@ public void init() {
...
+                configurationWrapper.getProp(ConfKeys.KEYS_EVENTMESH_TRACE_EXPORT_ZIPKIN_PORT);
             if (StringUtils.isNotEmpty(eventMeshTraceExportZipkinPortStr)) {
                 eventMeshTraceExportZipkinPort =
-                        Integer.parseInt(StringUtils.deleteWhitespace(eventMeshTraceExportZipkinPortStr));
+                    Integer.parseInt(StringUtils.deleteWhitespace(eventMeshTraceExportZipkinPortStr));
+            }
+
+            String metricsPluginType = configurationWrapper.getProp(ConfKeys.KEYS_EVENTMESH_METRICS_PLUGIN_TYPE);
+            if (StringUtils.isNotEmpty(metricsPluginType)) {
+                eventMeshMetricsPluginType = Arrays.stream(metricsPluginType.split(","))
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply eventmesh-common/src/main/java/org/apache/eventmesh/common/config/CommonConfiguration.java:179]: Some situalitions may require the use of multiple metrics plugins, so use List here.

File: eventmesh-metrics-plugin/eventmesh-metrics-api/src/main/java/org/apache/eventmesh/metrics/api/model/Metric.java [hunk 2]
```diff
@@ -15,20 +15,10 @@
...
-    private MetricRegistry metricRegistry;
-
-    public GroupMetrics(EventMeshHTTPServer eventMeshHTTPServer, MetricRegistry metricRegistry) {
-        this.eventMeshHTTPServer = eventMeshHTTPServer;
-        this.metricRegistry = metricRegistry;
-    }
+/**
+ * Top-level metric, all metrics should implement this interface.
+ */
+public interface Metric {
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply eventmesh-metrics-plugin/eventmesh-metrics-api/src/main/java/org/apache/eventmesh/metrics/api/model/Metric.java:23]: I suggest to use interface here, this class is just as a tag.

File: eventmesh-metrics-plugin/eventmesh-metrics-opentelemetry/src/main/java/org/apache/eventmesh/metrics/opentelemetry/OpenTelemetryMetricsRegistry.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    @Override
+    public void start() {
+        if (prometheusHttpServer == null) {
+            synchronized (OpenTelemetryMetricsRegistry.class) {
+                if (prometheusHttpServer == null) {
+                    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder().buildAndRegisterGlobal();
+                    PrometheusCollector
+                        .builder().setMetricProducer(sdkMeterProvider).buildAndRegister();
+                    // todo:
+                    int port = 0;
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply eventmesh-metrics-plugin/eventmesh-metrics-opentelemetry/src/main/java/org/apache/eventmesh/metrics/opentelemetry/OpenTelemetryMetricsRegistry.java]: Yes, you are right, I have modified this.

**Comment by Roc-00** [comment]
LGTM

File: eventmesh-metrics-plugin/eventmesh-metrics-opentelemetry/build.gradle [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+dependencies {
+    implementation project(":eventmesh-metrics-plugin:eventmesh-metrics-api")
+    implementation project(":eventmesh-common")
+    implementation 'org.slf4j:slf4j-api'
+    implementation 'org.apache.commons:commons-lang3'
+    implementation 'com.google.guava:guava'
+
+    // todo：Can we remove some dependency?
```
[the remark this answers is not in the document]
Comment (ruanwenjun) [reply eventmesh-metrics-plugin/eventmesh-metrics-opentelemetry/build.gradle:25]: This need to discuss, currently we use opentelemetry to export metrics, but we actually are using prometheus to export metrics. So the opentelemetry dependencied is not required, we may only need to use prometheus.

**Review by xwm1992 — APPROVED** [review]
LGTM

**Pull request merged**